.github/workflows/stage-build.yml

```
@@ -298,6 +298,51 @@ jobs:
         run: |-
           gsutil -m -h "Cache-Control:public, max-age=86400" rsync -r client/build gs://content-stage-mdn/main
 
+      # Deploy Function
+      - name: Authenticate with GCP
+        uses: google-github-actions/auth@v1
+        with:
+          token_format: access_token
+          service_account: deploy-stage-nonprod-mdn-ingre@${{ secrets.GCP_PROJECT_NAME }}.iam.gserviceaccount.com
+          workload_identity_provider: projects/${{ secrets.WIP_PROJECT_ID }}/locations/global/workloadIdentityPools/github-actions/providers/github-actions
+
+      - name: Setup gcloud
+        uses: google-github-actions/setup-gcloud@v1
+
+      - name: Build Function
+        working-directory: gcp/function
+        env:
+          CONTENT_ROOT: ${{ github.workspace }}/mdn/content/files
+          CONTENT_TRANSLATED_ROOT: ${{ github.workspace }}/mdn/translated-content/files
+        run: |
+          npm ci
+          npm run build
+
+      - name: Deploy Function
+        run: |-
+          for region in europe-west1 us-west1 asia-east1; do
+            gcloud functions deploy mdn-nonprod-stage-$region \
+            --gen2 \
+            --runtime=nodejs18 \
+            --region=$region \
+            --source=gcp/function \
+            --trigger-http \
+            --allow-unauthenticated \
+            --entry-point=mdnHandler \
+            --memory=256MB \
+            --timeout=60s \
+            --set-env-vars="ORIGIN_MAIN=developer.allizom.org" \
+            --set-env-vars="ORIGIN_LIVE_SAMPLES=yari-demos.developer.allizom.org" \
+            --set-env-vars="SOURCE_CONTENT=https://storage.googleapis.com/content-stage-mdn/" \
+            --set-env-vars="SOURCE_RUMBA=https://developer.allizom.org/" \
```

Actually, this should be a different host than `ORIGIN_MAIN`:

```suggestion
            --set-env-vars="SOURCE_RUMBA=https://api.developer.allizom.org/" \
```